sdk/identity/src/token_credentials/specific_azure_credential.rs

```diff
@@ -0,0 +1,316 @@
+use super::options;
+use std::sync::Arc;
+
+#[cfg(feature = "client_certificate")]
+use crate::ClientCertificateCredential;
+use crate::{
+    AppServiceManagedIdentityCredential, AzureCliCredential, ClientSecretCredential,
+    EnvironmentCredential, TokenCredentialOptions, VirtualMachineManagedIdentityCredential,
+    WorkloadIdentityCredential,
+};
+use azure_core::{
+    auth::{AccessToken, TokenCredential},
+    error::{ErrorKind, ResultExt},
+    Error,
+};
+
+pub const AZURE_CREDENTIAL_TYPE: &str = "AZURE_CREDENTIAL_TYPE";
+
+pub mod azure_credential_types {
+    pub const ENVIRONMENT: &str = "environment";
+    pub const AZURE_CLI: &str = "azurecli";
+    pub const VIRTUAL_MACHINE: &str = "virtualmachine";
+    pub const APP_SERVICE: &str = "appservice";
+    pub const CLIENT_SECRET: &str = "clientsecret";
+    pub const WORKLOAD_IDENTITY: &str = "workloadidentity";
+    #[cfg(feature = "client_certificate")]
+    pub const CLIENT_CERTIFICATE: &str = "clientcertificate";
+}
+
+/// Creates a `DefaultAzureCredential` by default with default options.
+/// If `AZURE_CREDENTIAL_TYPE` environment variable is set, it creates a `SpecificAzureCredential` with default options.
+pub fn create_credential() -> azure_core::Result<Arc<dyn TokenCredential>> {
```

Comment:
I am concerned with this new public API.

It's my understanding of the meeting we attended last week with the Azure Identity v-team, this doesn't align with their desires moving forward.  I'll try and wrap up what I took away from the meeting here:

1. The v-team "disliked" `DefaultAzureCredential` (DAC), but acknowledged DAC was going to stay around moving forward because it had the benefit to have a single, easy to cut and paste, authentication mechanism that "just worked" for examples and documentation.
2. DAC should behave similarly across all of the SDKs.
3. Adding a new undiscoverable public API to DAC should be avoided.  During the meeting, multiple people described using an environment variable as an undiscoverable public API.

As such, their recommendation was to _not_ add `SpecificAzureCredential` to DAC.

I acknowledge `create_credential` is not DAC by name, but it's DAC by intent. 
 It's _usually_ just DAC with a shorter name, and then put into all of the examples as it's easier to understand by name.

By adding `create_credential` as a public API, this effectively creates a rust-only DAC clone with the feature the v-team did not want included.

In general, the work in this PR includes multiple positive steps towards aligning the azure-identity in Rust with azure-identity in the other SDKs.  That said, I have concerns with `SpecificAzureCredential` and `create_credential` based on the feedback provided by the Azure Identity v-team.

Reply:
In the meeting, I proposed that `SpecificAzureCredential` functionality be added to `DefaultAzureCredential` across all Azure Identity SDKs. Their sentiment was to not add it to DAC. In commit 6c8edf71311f65cc31e592f5785fd17f099e9680, I removed `SpecificAzureCredential` from `DefaultAzureCredential` to align with that. It is a separate credential. Our DAC aligns with the other Azure SDKs.

This PR introduces 3 convenience factory method:
-  `create_default_credential` which creates an `Arc<DefaultAzureCredential>`
- `create_specific_credential` which creates an `Arc<SpecificAzureCredential>`
- `create_credential` which is a factory methods that will create one of the above two credentials

This aligns DAC with the other Azure SDKs and adds #1503.

Reply:
Why an `Arc<>` vs. a `Rc<>`, though? Protecting access to the whole thing seems unnecessary when it should support multiple readers, single writer. It only needs fully protecting when a token is refreshed.

Reply:
Arc<dyn TokenCredential> is what all the clients are created with. [search](https://github.com/search?q=repo%3AAzure%2Fazure-sdk-for-rust%20Arc%3Cdyn%20TokenCredential%3E&type=code)

Reply:
I get that. I guess this is more of a question overarching this discussion. I've opened #1571 to track and will not hold up this PR for that.